networks/archive/1b_corrected_correlation_analysis.py

```diff
@@ -0,0 +1,506 @@
+)
+plt.legend()
+plt.title("Variance explained for log transformed PA14")
+plt.ylabel("Variance ratio")
+plt.xlabel("Principal components")
+plt.show()
+# -
+
+# Correlation
+pao1_corr_pca = pao1_pc_weights_df.corr()
+pa14_corr_pca = pa14_pc_weights_df.corr()
+
+"""# Plot heatmap
+plt.figure(figsize=(20, 20))
+h3a = sns.clustermap(pao1_corr_pca.abs(), cmap="viridis")
+h3a.fig.suptitle(f"log transformed + PCA using {num_PCs} PCs (PAO1 genes)", y=1.05)"""
+
+"""plt.figure(figsize=(20, 20))
+h4a = sns.clustermap(pa14_corr_pca.abs(), cmap="viridis")
+h4a.fig.suptitle(f"log transformed + PCA using {num_PCs} PCs (PA14 genes)", y=1.05)"""
+
+# ## SEEK + correlation
+#
+# From [Zhu et. al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4768301/), we apply their "gene hubbiness correction" procedure. This is part of their [SEEK](https://seek.princeton.edu/seek/) algorithm. This correction procedure is motivated by the observation that hubby or well-connected genes in the co-expression network represent global, well-co-expressed processes, and can contaminate the search results regardless of query composition due to the effect of unbalanced gene connectivity in a scale-free co-expression network, and can lead to non-specific results in search or clustering approaches. To avoid the bias created by hubby genes they correct for each gene g’s correlation by subtracting g's average correlation.
+#
+# We noticed that the corrected correlation matrix is not symmetric anymore because we are subtracting the mean for each row. Instead we will normalize the correlation scores per gene
+
+# +
+# 0-1 normalize per gene
+scaler = preprocessing.MinMaxScaler()
+
+pao1_corr_scaled = scaler.fit_transform(pao1_corr_original)
+pa14_corr_scaled = scaler.fit_transform(pa14_corr_original)
+
+pao1_corr_seek = pd.DataFrame(
+    pao1_corr_scaled, columns=pao1_corr_original.columns, index=pao1_corr_original.index
+)
+pa14_corr_seek = pd.DataFrame(
+    pa14_corr_scaled, columns=pa14_corr_original.columns, index=pa14_corr_original.index
+)
+# -
+
+pao1_corr_original.head()
+
+pao1_corr_seek.head()
+
+"""# Plot heatmap
+plt.figure(figsize=(20, 20))
+h5 = sns.clustermap(pao1_corr_seek.abs(), cmap="viridis")
+h5.fig.suptitle("Correlation of PAO1 genes using (SEEK corrected)", y=1.05)"""
+
+"""plt.figure(figsize=(20, 20))
+h6 = sns.clustermap(pa14_corr_seek.abs(), cmap="viridis")
+h6.fig.suptitle("Correlation of PA14 genes (SEEK corrected)", y=1.05)"""
+
+# ## Hetio
+#
+# https://het.io/ allows users to search a heterogeneous network (i.e. are graphs with multiple node and edge types like disease or gene nodes) to find connections between entities such as eye and breast cancer. It is used as a hypothesis generating tool.
+#
+# To search the network, they empoloy a correction for hubbiness. Here we use their correction. This correction is similar to SEEK in that it is applying a correction using both the source and target node to deal with the asymmetry that we encountered.
```

Comment:
Interesting that this didn't seem to work for you!

Although maybe your problem isn't really specifically "hubbiness", since you're not looking at network paths like they were. Still, I might have expected it to help a bit with correcting for highly connected parts of the network.

Reply:
These curves should also be in [1b_corrected_correlation_analysis.ipynb](networks/archive/1b_corrected_correlation_analysis.ipynb)